config/routes.rb

```diff
@@ -324,7 +324,13 @@
       end
     end
     resources :comment_mutes, only: %i[update]
-    resources :users, only: %i[index], defaults: { format: :json } # internal API
+    resources :users, only: %i[index], defaults: { format: :json } do # internal API
+      constraints(-> { FeatureFlag.enabled?(:mobile_notifications) }) do
```

Comment:
Unrelated to this PR, but I wonder if we slowly want to wrap this up in a custom routing constraint like

```ruby
class FeatureFlagEnabled
  def initialize(flag)
    @flag = flag
  end

  def matches?(_request)
    FeatureFlag.enabled?(@flag)
  end
end
```

It would allow us to replace all these lambda constraints with things like 

```ruby
constraints(FeatureFlagEnabled.new(:mobile_notifications)) do
  ...
end

# or
resources :things, constraints: FeatureFlagEnabled.new(:things)
```

Reply:
I'm not going to tackle this in this PR, but I agree it's a good idea longer term